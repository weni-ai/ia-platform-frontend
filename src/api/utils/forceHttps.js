export default function forceHttps(receivedUrl) {
  const url = new URL(receivedUrl);

  url.protocol = 'https:';

  return url.toString();
}
