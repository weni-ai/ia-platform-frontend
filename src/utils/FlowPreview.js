// from https://github.com/weni-ai/floweditor/blob/main/src/components/simulator/Simulator.tsx

import axios from 'axios';
import update from 'immutability-helper';
import { v4 as createUUID } from 'uuid';

export const isMT = (event) => {
  return !!['msg_created', 'ivr_created'].find((type) => type === event.type);
};

export const isMessage = (event) => {
  return !!['msg_created', 'msg_received', 'ivr_created'].find(
    (type) => type === event.type,
  );
};

const MESSAGE_DELAY_MS = 200;

const DrawerType = {
  audio: 'audio',
  images: 'images',
  videos: 'videos',
  location: 'location',
  digit: 'digit',
  digits: 'digits',
  quickReplies: 'quickReplies',
};

export default {
  data() {
    return {
      preview: {
        flowUuid: '',

        contact: {
          fields: {},
          groups: [],
          urns: ['tel:+12065551212'],
          uuid: '29de4e4e-6cde-42ca-a96d-80fdeb373077',
        },

        active: false,
        events: [],
        quickReplies: [],

        context: {},
        sprinting: false,
        session: null,
        drawerOpen: false,
        drawerType: '',
        waitingForHint: false,
      },
    };
  },

  computed: {
    previewAPI() {
      const instance = axios.create({
        baseURL: runtimeVariables.get('FLOWS_API_BASE_URL'),
      });

      return (body) =>
        instance.post(`api/v2/flows/${this.preview.flowUuid}/simulate`, body, {
          headers: {
            Authorization: localStorage.getItem('authToken'),
          },
        });
    },
  },

  methods: {
    previewInit() {
      this.preview.contact = {
        uuid: createUUID(),
        urns: ['tel:+12065551212'],
        fields: {},
        groups: [],
        created_on: new Date().toISOString(),
      };
    },

    async previewStart({ languageId, flowUuid, flowName }) {
      this.preview.flowUuid = flowUuid;

      const now = new Date().toISOString();

      const contact = this.preview.contact;

      if (languageId) {
        contact.language = languageId;
      }

      this.sprinting = true;
      this.events = [];

      const body = {
        contact: this.preview.contact,
        trigger: {
          type: 'manual',
          environment: {
            date_format: 'DD-MM-YYYY',
            time_format: 'hh:mm',
            timezone: 'America/New_York',
            languages: [],
          },
          contact,
          flow: {
            uuid: flowUuid,
            name: flowName,
          },
          params: {},
          triggered_on: now,
        },
      };

      const response = await this.previewAPI(body);

      this.previewUpdateRunContext({
        ...response.data,
        events: JSON.parse(JSON.stringify(response.data.events)),
      });

      return response;
    },

    previewUpdateEvents(events, session, recentMessages, callback) {
      if (events && events.length > 0) {
        let quickReplies = null;

        let messageFound = false;
        while (events.length > 0 && !messageFound) {
          const event = events.shift();

          if (isMessage(event)) {
            messageFound = true;

            // if it's a message add it to our recent messages
            let fromUUID = '';
            let toUUID = '';

            // work backwards, since our events are recent
            for (let i = session.runs.length - 1; i >= 0; i--) {
              const path = session.runs[i].path;

              // start at the penultimate node since we have nowhere to render recent messages for the last node
              for (let j = path.length - 1; j >= 0; j--) {
                if (path[j].uuid === event.step_uuid) {
                  fromUUID = path[j].exit_uuid;
                  toUUID = path.length > j + 1 ? path[j + 1].node_uuid : null;
                  break;
                }
              }

              if (fromUUID) {
                const key = `${fromUUID}:${toUUID}`;
                const msg = {
                  sent: event.created_on,
                  text: event.msg.text,
                };
                if (key in recentMessages) {
                  recentMessages[key].unshift(msg);
                } else {
                  recentMessages[key] = [msg];
                }
              }
            }

            if (isMT(event)) {
              // save off any quick replies we might have
              if (event.msg.quick_replies) {
                quickReplies = event.msg.quick_replies;
              }
            }
          }

          this.preview.events.push(event);
        }

        this.preview.quickReplies = quickReplies;

        if (events.length === 0) {
          callback();
        } else {
          window.setTimeout(() => {
            this.previewUpdateEvents(events, session, recentMessages, callback);
          }, MESSAGE_DELAY_MS);
        }
      } else {
        callback();
      }
    },

    previewHasQuickReplies() {
      return (this.preview.quickReplies || []).length > 0;
    },

    previewUpdateRunContext(runContext, msg) {
      const wasJustActive =
        this.preview.active ||
        (runContext.events && runContext.events.length > 0);

      this.preview.quickReplies = [];

      if (!runContext.events || (runContext.events.length === 0 && msg)) {
        const runs = runContext.session.runs;
        const run = runs[runs.length - 1];
        const step = run.path[run.path.length - 1];

        runContext.events = [
          {
            msg: {
              uuid: createUUID(),
              urn: this.preview.contact.urns[0],
              text: msg.text,
              attachments: msg.attachments,
            },
            type: 'msg_created',
            created_on: new Date().toISOString(),
            step_uuid: step.uuid,
          },
        ];
      }

      const newlyRecentMessages = {};

      this.previewUpdateEvents(
        runContext.events,
        runContext.session,
        newlyRecentMessages,
        () => {
          const active = runContext.session.runs.some(
            ({ status }) => status === 'waiting',
          );

          if (!active && wasJustActive) {
            this.preview.events.push({
              type: 'info',
              text: 'Exited flow',
              created_on: new Date(),
            });
          }

          const waitingForHint =
            runContext.session &&
            runContext.session.wait &&
            runContext.session.wait.hint !== undefined;

          let drawerType = null;
          if (waitingForHint) {
            switch (runContext.session.wait.hint.type) {
              case 'audio':
                drawerType = DrawerType.audio;
                break;
              case 'video':
                drawerType = DrawerType.videos;
                break;
              case 'image':
                drawerType = DrawerType.images;
                break;
              case 'location':
                drawerType = DrawerType.location;
                break;
              case 'digits':
                drawerType = DrawerType.digit;
                if (runContext.session.wait.hint.count !== 1) {
                  drawerType = DrawerType.digits;
                }
                break;
              default:
                console.log('Unknown hint', runContext.session.wait.hint.type);
            }
          }

          let drawerOpen = waitingForHint;

          // if we have quick replies, open our drawe with attachment options
          if (!drawerType && this.previewHasQuickReplies()) {
            drawerType = DrawerType.quickReplies;
            drawerOpen = true;
          }

          this.preview.active = active;
          this.preview.context = runContext.context;
          this.preview.sprinting = false;
          this.preview.session = runContext.session;
          this.preview.drawerOpen = drawerOpen;
          this.preview.drawerType = drawerType;
          this.preview.waitingForHint = waitingForHint;

          this.previewUpdateActivity(newlyRecentMessages);
        },
      );
    },

    previewUpdateActivity(recentMessages) {
      if (this.preview.session) {
        // if we are resetting, clear our recent messages

        let lastExit = null;
        const paths = {};
        const active = {};
        let activeFlow = '';

        for (const run of this.preview.session.runs) {
          let finalStep = null;

          for (const step of run.path) {
            if (lastExit) {
              const key = lastExit + ':' + step.node_uuid;
              let pathCount = paths[key];
              if (!pathCount) {
                pathCount = 0;
              }
              paths[key] = ++pathCount;
              if (!(key in recentMessages)) {
                recentMessages[key] = [];
              }
            }
            lastExit = step.exit_uuid;
            finalStep = step;
          }

          if (finalStep) {
            let count = active[finalStep.node_uuid];
            if (!count) {
              count = 0;
            }

            if (lastExit) {
              const lastKey = lastExit + ':' + null;
              paths[lastKey] = 1;

              if (!(lastKey in recentMessages)) {
                recentMessages[lastKey] = [];
              }
            }

            if (this.preview.session.status === 'waiting') {
              active[finalStep.node_uuid] = ++count;
            }
            activeFlow = run.flow_uuid;
          }
        }
      }
    },

    async previewResume(text) {
      if (!text) {
        return;
      }

      this.preview.sprinting = true;

      const now = new Date().toISOString();

      const msg = {
        text,
        uuid: createUUID(),
        urn: this.preview.session.contact.urns[0],
        attachments: [],
      };

      const body = {
        session: this.preview.session,
        resume: {
          type: 'msg',
          msg,
          resumed_on: now,
          contact: this.preview.session.contact,
        },
      };

      try {
        const response = await this.previewAPI(body);

        await this.previewUpdateRunContext(
          {
            ...response.data,
            events: JSON.parse(JSON.stringify(response.data.events)),
          },
          msg,
        );

        return response;
      } catch (error) {
        const events = update(this.preview.events, {
          $push: [
            {
              type: 'error',
              text:
                error.response.status > 499
                  ? 'Server error, try again later'
                  : error.response.data.error,
            },
          ],
        });

        this.preview.events = events;
      }
    },
  },
};
