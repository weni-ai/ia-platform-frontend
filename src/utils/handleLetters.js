export function lowerDirstCapitalLetter(text) {
  if (!text) return '';

  return text.charAt(0).toLowerCase() + text.slice(1);
}
