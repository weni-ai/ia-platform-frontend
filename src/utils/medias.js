export const allowedMediaFormats = {
  image: ['.png', '.jpeg', '.jpg', '.gif', '.webp'],
  video: ['.mp4', '.avi'],
  document: ['.doc', '.docx', '.xls', '.xlsx', '.pdf', '.txt'],
};

export function getFileExtension(file) {
  const getExtension = (name) => {
    const parts = name.split('.');
    return parts.length > 1 ? `.${parts.pop().toLowerCase()}` : null;
  };

  if (file && file.name) {
    return getExtension(file.name);
  }

  return null;
}

export function getFileType(src) {
  const isImage = () => {
    const extension = getFileExtension(src);
    return extension && allowedMediaFormats.image.includes(extension);
  };

  const isVideo = () => {
    const extension = getFileExtension(src);
    return extension && allowedMediaFormats.video.includes(extension);
  };

  const isGeolocation = () => {
    const geolocationRegex =
      /^([-+]?([1-8]?\d(\.\d+)?|90(\.0+)?)),\s*([-+]?((1[0-7]\d(\.\d+)?)|([1-9]?\d(\.\d+)?|180(\.0+)?)))$/;

    return geolocationRegex.test(src);
  };

  const isDocument = () => {
    const extension = getFileExtension(src);
    return extension && allowedMediaFormats.document.includes(extension);
  };

  const typeMap = {
    image: isImage,
    video: isVideo,
    geolocation: isGeolocation,
    document: isDocument,
  };

  const fileType = Object.entries(typeMap).find(([_type, check]) =>
    check(),
  )?.[0];

  return fileType;
}

export function formatFileSize(size) {
  const units = ['B', 'KB', 'MB', 'GB'];
  let unitIndex = 0;

  while (size >= 1024 && unitIndex < units.length - 1) {
    size /= 1024;
    unitIndex++;
  }

  return `${Math.round(size)} ${units[unitIndex]}`;
}
