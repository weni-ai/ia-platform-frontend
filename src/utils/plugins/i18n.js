import * as VueI18n from 'vue-i18n';

import en from '../../locales/en.json';
import ptbr from '../../locales/pt_br.json';
import es from '../../locales/es.json';

const languages = {
  'en-US': en,
  'pt-BR': ptbr,
  es,
};

const numberFormats = {
  'en-US': {
    currency: {
      style: 'currency',
      currency: 'BRL',
      currencyDisplay: 'symbol',
    },
  },
  'pt-BR': {
    currency: {
      style: 'currency',
      currency: 'BRL',
      currencyDisplay: 'symbol',
    },
  },
  es: {
    currency: {
      style: 'currency',
      currency: 'BRL',
      currencyDisplay: 'symbol',
    },
  },
};

const messages = Object.assign(languages);

const i18n = VueI18n.createI18n({
  locale: navigator.language || navigator.userLanguage,
  fallbackLocale: 'en-US',
  messages,
  numberFormats,
});

export default i18n;
