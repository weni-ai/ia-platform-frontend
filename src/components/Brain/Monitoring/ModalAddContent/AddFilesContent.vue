<template>
  <section class="add-files-content">
    <section
      v-if="files.data.length"
      class="add-files-content__files"
    >
      <ContentItem
        v-for="file in files.data"
        :key="file.uuid"
        :file="file"
        :compressed="true"
        :clickable="true"
        @remove="remove(file)"
      />
    </section>
    <template v-else>
      <input
        v-show="false"
        ref="browser-file-input"
        type="file"
        :accept="allowedFormats"
        multiple
        @change="inputFile"
      />
      <StartAddContent
        :description="$t('content_bases.files.description')"
        :subDescription="removeHTML(supportedFormats)"
        :textAddContent="$t('content_bases.files.browse_file')"
        @add="$refs['browser-file-input'].click()"
      />
    </template>
  </section>
</template>

<script setup>
import { computed, reactive, watch } from 'vue';
import { useStore } from 'vuex';
import { get } from 'lodash';

import nexusaiAPI from '@/api/nexusaiAPI';
import i18n from '@/utils/plugins/i18n';
import { allowedMediaFormats, fileSizeLimitInMegabytes } from '@/utils/medias';

import StartAddContent from './StartAddContent.vue';
import ContentItem from '../../ContentBase/ContentItem.vue';
import { useFilesPagination } from '@/views/ContentBases/filesPagination';

const emit = defineEmits(['update:model-value']);

const store = useStore();
const contentBaseUuid = computed(() => store.state.router.contentBaseUuid);
const allowedFormats = allowedMediaFormats.document;
const files = reactive(
  useFilesPagination({
    contentBaseUuid: contentBaseUuid.value,
  }),
);

watch(files.data, (newFilesData) => {
  console.log('newFilesData', newFilesData);

  emit('update:model-value', newFilesData);
});

function removeHTML(string) {
  const span = document.createElement('span');
  span.innerHTML = string;
  return span.textContent;
}

const supportedFormats = computed(() => {
  return i18n.global.t('content_bases.files.sup_files', {
    suportedFormats: [
      allowedFormats.slice(0, -1).join(', '),
      allowedFormats.at(-1),
    ].join(i18n.global.t('and')),
    limitMB: fileSizeLimitInMegabytes,
  });
});

function inputFile(event) {
  const files =
    get(event, 'dataTransfer.files') || get(event, 'srcElement.files');

  if (!get(files, 'length', 0)) {
    return;
  }

  Object.values(files).forEach(addFile);

  const srcElement = get(event, 'srcElement');

  srcElement.value = '';
}

function addFile(file) {
  const extension =
    file.name.lastIndexOf('.') === -1
      ? file.name
      : file.name.slice(file.name.lastIndexOf('.') + 1);

  const isNotAllowedFormat = !allowedFormats.includes(
    `.${extension}`.toLowerCase(),
  );

  if (isNotAllowedFormat) {
    return (store.state.alert = {
      type: 'error',
      text: i18n.global.t('content_bases.files.unsupported_format'),
    });
  }

  const fileSizeInMegabytes = file.size / Math.pow(1024, 2);

  if (fileSizeInMegabytes > fileSizeLimitInMegabytes) {
    return (store.state.alert = {
      type: 'error',
      text: i18n.global.t('content_bases.files.exceeds_limit', {
        limitMB: fileSizeLimitInMegabytes,
      }),
    });
  }

  const fileItem = reactive({
    uuid: `temp-${Math.floor(Math.random() * 1e9)}`,
    file: file.name,
    created_file_name:
      extension !== file.name
        ? file.name.slice(0, -extension.length - 1)
        : file.name,
    extension_file: extension,
    status: 'waiting',
    progress: 0,
    file_name: file.name,
  });

  files.addItem(fileItem);

  fileItem.status = 'uploading';

  nexusaiAPI.intelligences.contentBases.files
    .create({
      contentBaseUuid: contentBaseUuid.value,
      file,
      extension_file: extension,

      onUploadProgress: (event) => {
        fileItem.progress = event.loaded / event.total;
      },
    })
    .then(({ data }) => {
      fileItem.uuid = data.uuid;
      fileItem.extension_file = data.extension_file;
      fileItem.progress = 0;
      fileItem.status = 'processing';
    })
    .catch(() => {
      fileItem.status = 'fail-upload';
      fileItem.file_name = file.name;
    });
}

function remove(file) {
  nexusaiAPI.intelligences.contentBases.files
    .delete({
      contentBaseUuid: contentBaseUuid.value,
      fileUuid: file.uuid,
    })
    .then(() => {
      this.alert = {
        type: 'default',
        text: i18n.global.t('content_bases.files.file_removed_from_base', {
          name: file.name,
        }),
      };

      files.removeItem({ uuid: file.uuid });
    });
}
</script>

<style scoped lang="scss">
.add-files-content {
  height: 100%;

  display: flex;
  justify-content: center;

  &__files {
    width: 100%;

    display: flex;
    flex-direction: column;
    gap: $unnnic-spacing-xs;
  }
}
</style>
