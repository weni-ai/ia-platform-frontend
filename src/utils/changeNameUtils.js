import i18n from '@/utils/plugins/i18n';

function parseActionDetails(actionDetails) {
  return Object.keys(actionDetails).map((key) => ({
    key,
    newValue: actionDetails[key]?.new,
    oldValue: actionDetails[key]?.old,
  }));
}

function generateActionUpdateText(actionDetails) {
  const actionUpdateName = actionDetails[0]?.key === 'name' ? 'name' : 'prompt';
  const moreThanOneChange =
    actionDetails.length > 1 && actionDetails[0] !== 'new';

  if (moreThanOneChange)
    return i18n.global.t(`router.tunings.history.fields.changes`, {
      value: actionDetails.length,
    });

  return i18n.global.t(
    `router.tunings.history.fields.update-${actionUpdateName}-action`,
    {
      value: actionDetails[0]?.newValue,
    },
  );
}

function generateCustomizationUpdateText(actionDetails) {
  const moreThanOneChange =
    actionDetails.length > 1 && actionDetails[0] !== 'new';
  const isCustomization = [
    'name',
    'goal',
    'role',
    'personality',
    'instruction',
  ].includes(actionDetails[0]?.key);

  if (moreThanOneChange)
    return i18n.global.t(`router.tunings.history.fields.changes`, {
      value: actionDetails.length,
    });

  if (isCustomization)
    return i18n.global.t(
      `router.tunings.history.fields.update-${actionDetails[0]?.key}`,
      {
        value: actionDetails[0]?.newValue,
      },
    );

  return i18n.global.t(`router.tunings.history.fields.changes`);
}

function handleGroupText(actionDetails, isAction = false) {
  const translationName = (name) =>
    isAction
      ? `router.tunings.history.fields.update-${name}-action`
      : `router.tunings.history.fields.update-${name}`;
  const condition = isAction
    ? ['name', 'prompt']
    : ['name', 'goal', 'role', 'personality', 'instruction'];

  if (
    actionDetails.length > 1 &&
    !['new', 'old'].includes(actionDetails.map((e) => e.key))
  )
    return actionDetails.map((e) => {
      if (condition.includes(e.key))
        return i18n.global.t(translationName(e.key), {
          value: e.newValue,
        });

      return '';
    });

  return [];
}

function handleChangeName(row) {
  if (
    !row?.model_group &&
    row.action_details &&
    row?.action_details['brain_on']
  ) {
    const statusBrain = JSON.parse(
      row.action_details['brain_on'].new.toLowerCase(),
    )
      ? 'on'
      : 'off';

    return {
      icon: 'settings',
      user: row.created_by,
      text: i18n.global.t(`router.tunings.history.fields.brain-${statusBrain}`),
    };
  }

  if (!row || !row.model_group || !row.action_type || !row.action_details) {
    return {
      icon: 'article',
      user: '-',
      text: '-',
    };
  }

  const actionDetails = parseActionDetails(row.action_details);

  const groupData = {
    Action: {
      C: {
        icon: 'bolt',
        user: row.created_by,
        text: i18n.global.t('router.tunings.history.fields.add-action', {
          value: row.action_details.new,
        }),
      },
      U: {
        icon: 'bolt',
        user: row.created_by,
        text: generateActionUpdateText(actionDetails),
        groupText: handleGroupText(actionDetails, true),
      },
      D: {
        icon: 'bolt',
        user: row.created_by,
        text: i18n.global.t('router.tunings.history.fields.remove-action', {
          value: row.action_details.old,
        }),
      },
    },
    Config: {
      U: {
        icon: 'settings',
        user: row.created_by,
        text: i18n.global.t('router.tunings.history.fields.update-model'),
      },
    },
    Content: {
      D: {
        icon: 'article',
        user: row.created_by,
        text: i18n.global.t('router.tunings.history.fields.remove-content', {
          value: row.action_details.new,
        }),
      },
      U: {
        icon: 'article',
        user: row.created_by,
        text: i18n.global.t('router.tunings.history.fields.update-content', {
          value: row.action_details.new,
        }),
      },
      C: {
        icon: 'article',
        user: row.created_by,
        text: i18n.global.t('router.tunings.history.fields.add-content', {
          value: row.action_details.new,
        }),
      },
    },
    Customization: {
      U: {
        icon: 'person',
        user: row.created_by,
        text: generateCustomizationUpdateText(actionDetails),
        groupText: handleGroupText(actionDetails),
      },
    },
  };

  return groupData[row.model_group]?.[row.action_type];
}

export { handleChangeName };
