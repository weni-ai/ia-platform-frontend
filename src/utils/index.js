import { useActionsStore } from '@/store/Actions';
import VERBOSE_LANGUAGES from './verbose_languages';
import i18n from './plugins/i18n';

export const languageListToDict = (list) =>
  list.reduce((current, lang) => {
    Object.assign(current, { [lang]: VERBOSE_LANGUAGES[lang] || lang });
    return current;
  }, {});

export const LANGUAGES = languageListToDict(
  runtimeVariables
    .get('VITE_SUPPORTED_LANGUAGES')
    .split('|')
    .map((v) => v.split(':')[0]),
);

export const WENIGPT_OPTIONS = JSON.parse(
  runtimeVariables.get('VITE_OPTIONS_WENIGPT'),
);

export const BRAIN_ROUTES = [
  {
    title: 'monitoring',
    description: i18n.global.t('router.monitoring.description'),
    page: 'router-monitoring',
    icon: 'bar_chart',
    preview: false,
  },
  {
    title: 'personalization',
    page: 'router-personalization',
    icon: 'person',
    preview: true,
  },
  { title: 'content', page: 'router-content', icon: 'article', preview: true },
  {
    title: 'actions',
    description: i18n.global.t('router.actions.description'),
    page: 'router-actions',
    icon: 'bolt',
    preview: true,
  },
  {
    title: 'tunings',
    page: 'router-tunings',
    icon: 'settings',
    preview: true,
  },
];

export const createDownloadAnchor = ({ name, href }) => {
  const a = document.createElement('a');

  a.setAttribute('download', name);
  a.setAttribute('href', href);

  return a;
};

export const actionGroupIcon = (groupId) => {
  return {
    interactions: 'chat',
    shopping: 'shopping_cart',
    support: 'contact_support',
    media: 'attach_file',
    custom: 'edit_square',
  }[groupId];
};

export const actionInfo = ({ name, prompt, type }) => {
  const actionsStore = useActionsStore();

  const actionType = actionsStore.types.data.find(
    (item) =>
      item.name === name && item.prompt === prompt && item.type === type,
  );

  const group = actionType?.group || 'custom';

  return {
    icon: actionGroupIcon(group),
    group,
  };
};
