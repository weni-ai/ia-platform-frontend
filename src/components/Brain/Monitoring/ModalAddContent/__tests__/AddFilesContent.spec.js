import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import { describe, it, expect, vi, beforeEach } from 'vitest';

import { fileSizeLimitInMegabytes } from '@/utils/medias';
import i18n from '@/utils/plugins/i18n';

import AddFilesContent from '../AddFilesContent.vue';

vi.mock('@/api/nexusaiAPI', () => ({
  default: {
    intelligences: {
      contentBases: {
        files: {
          create: vi.fn(),
          delete: vi.fn(),
        },
      },
    },
  },
}));

vi.mock('@/utils/medias', () => ({
  allowedMediaFormats: { document: ['.pdf', '.doc', '.docx'] },
  fileSizeLimitInMegabytes: 50,
}));

const store = createStore({
  state: {
    router: { contentBaseUuid: '123' },
    alert: null,
  },
});

describe('AddFilesContent.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(AddFilesContent, {
      global: {
        plugins: [store, i18n],
      },
    });
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders StartAddContent when there are no files', () => {
    expect(wrapper.find('[data-testid="start-add-content"]').exists()).toBe(
      true,
    );
  });

  it('renders ContentItem for each file when files are present', async () => {
    wrapper.vm.files.data = [
      { uuid: 'file-1', file: 'example.pdf' },
      { uuid: 'file-2', file: 'example.docx' },
    ];

    await wrapper.vm.$nextTick();

    const contentItems = wrapper.findAllComponents(
      '[data-testid="content-item"]',
    );
    expect(contentItems).toHaveLength(2);
  });

  it('emits "update:model-value" when files.data changes', async () => {
    wrapper.vm.files.data.push({ uuid: 'file-1', file: 'example.pdf' });
    await wrapper.vm.$nextTick();

    expect(wrapper.emitted('update:model-value')).toBeTruthy();
  });

  it('triggers file input click when StartAddContent emits "add"', async () => {
    const input = wrapper.find('[data-testid="input-file"]');
    const clickSpy = vi.spyOn(input.element, 'click');

    wrapper.findComponent('[data-testid="start-add-content"]').vm.$emit('add');
    expect(clickSpy).toHaveBeenCalled();
  });

  it('displays an alert for unsupported file formats', async () => {
    const input = wrapper.find('[data-testid="input-file"]');
    const file = new File(['test'], 'example.exe', { type: 'application/exe' });

    Object.defineProperty(input.element, 'files', {
      value: [file],
      writable: false,
    });
    await input.trigger('change');

    expect(store.state.alert.text).toContain(
      i18n.global.t('content_bases.files.unsupported_format'),
    );
  });

  it('displays an alert for files exceeding size limit', async () => {
    const input = wrapper.find('[data-testid="input-file"]');
    const file = new File(['test'], 'largefile.pdf', {
      type: 'application/pdf',
    });
    Object.defineProperty(file, 'size', {
      value: (fileSizeLimitInMegabytes + 1) * Math.pow(1024, 2),
    });

    Object.defineProperty(input.element, 'files', {
      value: [file],
      writable: false,
    });
    await input.trigger('change');

    expect(store.state.alert.text).toContain(
      i18n.global.t('content_bases.files.exceeds_limit', {
        limitMB: fileSizeLimitInMegabytes,
      }),
    );
  });

  it('removes a file when ContentItem emits "remove"', async () => {
    const mockFile = { uuid: 'file-1', file: 'example.pdf' };
    wrapper.vm.files.data = [mockFile];

    await wrapper.vm.$nextTick();

    const spyRemove = vi
      .spyOn(wrapper.vm, 'remove')
      .mockImplementation(() => {});
    const contentItem = wrapper.findComponent('[data-testid="content-item"]');

    contentItem.vm.$emit('remove', mockFile);

    expect(spyRemove).toHaveBeenCalledWith(mockFile);
  });
});
