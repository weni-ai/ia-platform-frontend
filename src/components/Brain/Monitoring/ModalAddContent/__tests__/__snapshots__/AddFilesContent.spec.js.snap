// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddFilesContent.vue > matches the snapshot 1`] = `
"<section data-v-99057710="" class="add-files-content"><input data-v-99057710="" data-testid="input-file" type="file" accept=".pdf,.doc,.docx" multiple="" style="display: none;">
  <start-add-content-stub data-v-99057710="" description="Upload documents and files that the agent will use to provide answers" subdescription=".pdf, .doc and .docx. 50MB limit per file." icon="text_snippet" iconfilled="true" textaddcontent="Add file" data-testid="start-add-content"></start-add-content-stub>
</section>"
`;
